server/queue/queue-jms/src/test/java/org/apache/james/queue/jms/JMSMailQueueTest.java

```diff
@@ -49,7 +49,8 @@ public void setUp(BrokerService broker) throws Exception {
         RawMailQueueItemDecoratorFactory mailQueueItemDecoratorFactory = new RawMailQueueItemDecoratorFactory();
         NoopMetricFactory metricFactory = new NoopMetricFactory();
         String queueName = BrokerExtension.generateRandomQueueName(broker);
-        mailQueue = new JMSMailQueue(connectionFactory, mailQueueItemDecoratorFactory, queueName, metricFactory);
+        MailQueue.ConsumerOptions consumerOptions = name -> name + "?consumer.prefetchSize=0";
```

Comment:
I should have an explicit API:

```
ConsumerOptions.builder()
   .noPrefetchSize()
   .build();
```

And not rely on strings at the API level.

Reply:
Does every JMS implementation support prefetch ? `ConsumerOptions` is a general propose interface...so we should use only features that supported by the JMS spec.